public/app/plugins/panel/piechart/PieChartPanel.tsx

```diff
@@ -92,7 +92,7 @@ function getLegend(props: Props, displayValues: FieldDisplay[]) {
       } else if (isNaN(b.display.numeric)) {
         return -1;
       } else {
-        return b.display.numeric - a.display.numeric;
+        return (a.colIndex ?? Infinity) - (b.colIndex ?? Infinity);
```

Comment:
hey @SchneiderMaster, i don't think we want to replace numeric sorting entirely. the idea here is probably to respect colIndex when it is there, but otherwise continue sorting by numeric value?